src/updater/updater.cpp

```diff
@@ -149,446 +95,568 @@ void CheckWeightsConsistency(
+                       });
+
+    using WeightAndDuration = std::tuple<EdgeWeight, EdgeWeight>;
+    const auto compute_new_weight_and_duration =
+        [&](const GeometryID geometry_id) -> WeightAndDuration {
+        EdgeWeight new_weight = 0;
+        EdgeWeight new_duration = 0;
+        if (geometry_id.forward)
         {
-            using extractor::lookup::SegmentHeaderBlock;
-            using extractor::lookup::SegmentBlock;
-
-            auto header = reinterpret_cast<const SegmentHeaderBlock *>(edge_segment_byte_ptr);
-            BOOST_ASSERT(is_aligned<SegmentHeaderBlock>(header));
-            edge_segment_byte_ptr += sizeof(SegmentHeaderBlock);
-
-            auto first = reinterpret_cast<const SegmentBlock *>(edge_segment_byte_ptr);
-            BOOST_ASSERT(is_aligned<SegmentBlock>(first));
-            edge_segment_byte_ptr += sizeof(SegmentBlock) * (header->num_osm_nodes - 1);
-            auto last = reinterpret_cast<const SegmentBlock *>(edge_segment_byte_ptr);
-
-            // Find a segment with zero speed and simultaneously compute the new edge weight
-            EdgeWeight new_weight = 0;
-            EdgeWeight new_duration = 0;
-            auto osm_node_id = header->previous_osm_node_id;
-            bool skip_edge =
-                std::find_if(first, last, [&](const auto &segment) {
-                    auto segment_weight = segment.segment_weight;
-                    auto segment_duration = segment.segment_duration;
-                    if (auto value = segment_speed_lookup({osm_node_id, segment.this_osm_node_id}))
-                    {
-                        // If we hit a 0-speed edge, then it's effectively not traversible.
-                        // We don't want to include it in the edge_based_edge_list.
-                        if (value->speed == 0)
-                            return true;
-
-                        segment_duration = ConvertToDuration(segment.segment_length, value->speed);
-
-                        segment_weight =
-                            ConvertToWeight(value->weight, weight_multiplier, segment_duration);
-                    }
-
-                    // Update the edge weight and the next OSM node ID
-                    osm_node_id = segment.this_osm_node_id;
-                    new_weight += segment_weight;
-                    new_duration += segment_duration;
-                    return false;
-                }) != last;
-
-            // Update the node-weight cache. This is the weight of the edge-based-node only,
-            // it doesn't include the turn. We may visit the same node multiple times, but
+            const auto weights = segment_data.GetForwardWeights(geometry_id.id);
+            for (const auto weight : weights)
+            {
+                if (weight == INVALID_EDGE_WEIGHT)
+                {
+                    new_weight = INVALID_EDGE_WEIGHT;
+                    break;
```

Comment:
Yeah if one segment on the path is blocked we can't fully traverse it anymore, hence we filter it out completely.
If we should really remove the whole edge is up to debate but this is consistent with old behavior.